### DISCUSSION TITLE
Dynamic injected value doesn't work after upgrading to 3.3.4

### DISCUSSION THREAD

**Discussion by bruceontheloose** [statamic/cms#5762] [Troubleshooting] [answered]
After upgrading from 3.2.37 to 3.3.4, I'm experiencing an error with an injected value. So you know, I didn't opt in to the new Antlers parser.

Here's the relevant bit from our collection's config:

```yaml
inject:
  title_as_month_day: '{{title insert="2020-|0" format="F j"}}'
```

Entry titles in this collection are in the form of `mm-dd`, so for the entry with `title: 10-02`, the injected `title_as_month_day` would have a value of "October 2," which we render in various spots in the page.

That all worked fine prior to upgrading. But now `title_as_month_day` renders as `{{title insert="2020-|0" format="F j"}}`. To test whether `insert` or `format` were part of the issue, I tried injecting `title_as_month_day: '{{title}}'`. I got a similar result: `title_as_month_day` rendered as `{{title}}` instead of the value. 

In case you're curious, the same problem occurs with the new Antlers parser enabled.

Is there a solution you're aware of or should I file this as a bug?

Thanks.

**Comment by jasonvarga** [comment, marked as answer]
I wouldn't have thought you could put Antlers in injected variables like that in the first place. I'm surprised it ever worked.

I'd suggest removing that variable and making a tag instead.

```php
<?php

namespace App\Tags;

use Carbon\Carbon;
use Statamic\Tags\Tags;

class TitleAsMonthDay extends Tags
{
    public function index()
    {
        $title = '2020-'.$this->context->get('title');
        return Carbon::parse($title)->format('F j');
    }
}
```

**Comment by bruceontheloose** [comment]
Thanks so much, @jasonvarga! I really appreciate it.

I was new to creating custom tags, so here are the steps I took to implement Jason's code, for anyone who may land here and find them useful. (Docs page for reference: [Building Tags](https://statamic.dev/extending/tags).)

1. Ran `php please make:tag TitleAsMonthDay`, which added the stub file, `/app/Tags/TitleAsMonthDay.php`.
2. Replaced all the starter code in `TitleAsMonthDay.php` with Jason's code.

From there, you can use the tag in a template with `{{ title_as_month_day }}`. (As the docs say, "the tag’s “handle”...will be the snake_cased version of the class name by default.") In our case, I'm passing it into a partial, similar to:

```php
{{ 
  partial:example_partial 
    example_date_text="{title_as_month_day}"
}}
```

Thanks, again!
